mem0/vector_stores/azure_ai_search.py

```diff
@@ -195,7 +234,8 @@ def update(self, vector_id, vector=None, payload=None):
         self.search_client.merge_or_upload_documents(documents=[document])
```

Comment:
I don't remember if `merge_or_upload_documents` throws an exception on failure, since it could be a partial success case. The return value has a collection with per-document status, you may need to check that and throw an exception if the single doc in the batch failed. Same for the delete case.